### ISSUE TITLE
Clean up rounding mode support

### ISSUE THREAD

**Issue by nemequ** [nemequ/simde#347]
The rounding mode support is a bit of a mess right now and doesn't really work properly.  Several of the tests actually change the rounding mode to truncation to make it consistent, which is okay (except we should also test the other modes).

On ARMv8+ NEON doesn't actually use the current rounding mode (the FPSCR register); see the [Arm® Architecture Reference Manual](https://static.docs.arm.com/ddi0487/fb/DDI0487F_b_armv8_arm.pdf) docs on FPSCR.  However, autovectorization of __builtin_nearbyintf on GCC [will compile to FRINTI](https://godbolt.org/z/8momHa).

I'm thinking it might be best to move _mm_cvtps_epi32 from sse2.h to sse.h, but the code in that function also needs to be improved to use the current rounding mode instead of just assuming round-to-nearest with ties toward even.

We should probably be using `simde_math_nearbyintf` on most systems, except for NEON on ARMv8+ where we'll need a big switch on fegetenv() for non-x86.

Properly handling ties should also probably be optional, which brings us back to a -ffast-math type macro like the one planned for NaNs.  I'm thinking maybe it should be a bitmask with a few different options. So far:

 * Fast vs. accurate NaNs
 * Fast vs. accurate rounding [i.e., should we actually bother to check fegetenv()])
   * Fast vs. accurate rounding of ties

Basically, it's a solvable problem, someone just needs to spend a bit of time cleaning everything up.  I'm pretty sure that person is going to be me, but I don't want to forget so I'm creating this issue.

**Comment by nemequ** [comment]
Done.  Turns out it's not as bad as I'd feared; they ignore FPSCR, but it uses a different register (FPCR).

**Comment by p0nce** [comment]
Here is how it works in `intel-intrinsics`:

- since we only support x86 as input API, `_MM_SET_ROUNDING_MODE` is the one entry point to change the rounding mode, and all 4 are supported.
- `_MM_SET_ROUNDING_MODE` changes MXCSR, on arm64/arm32 it changes FPSCR/FPCR (resp.). Current rounding mode is store there because it's a thread local place. Obviously for other target arch you would need a register to store things in. MXCSR is emulated but we don't support all features in MXCSR, like FP exceptions states/mask.
- any operation that converts using the "current" rounding mode gets it from MXCSR / FPSCR / FPCR
- we resort to libc for arm32 rounding
- In the case of 32-bit x86 there is no SSE2 way to convert floating-point to 64-bit int. We then do an annoying sequence to save/restore the FPU rounding mode and take that from SSE just for the conversion.

I don't imagine the complexity with several input APIs.

That said, this "current" rounding mode is annoying for modern arm64 and x86_64, since you can have explicit rounding-mode all the time. It is at times inefficient because you have to switch on the current rounding mode and dispatch to different neon builtins.
